src/Microsoft.AspNetCore.Routing/Internal/ParameterPolicyActivator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Routing.Internal
+{
+    internal static class ParameterPolicyActivator
+    {
+        public static T ResolveParameterPolicy<T>(IDictionary<string, Type> inlineParameterPolicyMap, IServiceProvider serviceProvider, string inlineParameterPolicy, out string parameterPolicyKey)
+            where T : IParameterPolicy
+        {
+            if (inlineParameterPolicyMap == null)
+            {
+                throw new ArgumentNullException(nameof(inlineParameterPolicyMap));
+            }
+
+            if (inlineParameterPolicy == null)
+            {
+                throw new ArgumentNullException(nameof(inlineParameterPolicy));
+            }
+
+            string argumentString;
+            var indexOfFirstOpenParens = inlineParameterPolicy.IndexOf('(');
+            if (indexOfFirstOpenParens >= 0 && inlineParameterPolicy.EndsWith(")", StringComparison.Ordinal))
+            {
+                parameterPolicyKey = inlineParameterPolicy.Substring(0, indexOfFirstOpenParens);
+                argumentString = inlineParameterPolicy.Substring(
+                    indexOfFirstOpenParens + 1,
+                    inlineParameterPolicy.Length - indexOfFirstOpenParens - 2);
+            }
+            else
+            {
+                parameterPolicyKey = inlineParameterPolicy;
+                argumentString = null;
+            }
+
+            if (!inlineParameterPolicyMap.TryGetValue(parameterPolicyKey, out var parameterPolicyType))
+            {
+                return default;
+            }
+
+            if (!typeof(T).IsAssignableFrom(parameterPolicyType))
+            {
+                throw new RouteCreationException(
+                            Resources.FormatDefaultInlineConstraintResolver_TypeNotConstraint(
+                                                        parameterPolicyType, parameterPolicyKey, typeof(T).Name));
+            }
+
+            try
+            {
+                return (T)CreateParameterPolicy(serviceProvider, parameterPolicyType, argumentString);
+            }
+            catch (RouteCreationException)
+            {
+                throw;
+            }
+            catch (Exception exception)
+            {
+                throw new RouteCreationException(
+                    $"An error occurred while trying to create an instance of '{parameterPolicyType.FullName}'.",
+                    exception);
+            }
+        }
+
+        private static IParameterPolicy CreateParameterPolicy(IServiceProvider serviceProvider, Type constraintType, string argumentString)
+        {
+            var constraintTypeInfo = constraintType.GetTypeInfo();
```

Comment:
This  should just be `GetType` - all the type info stuff is legacy from the bad old days 😆